client/src/routes/profile.svelte

```diff
@@ -0,0 +1,28 @@
+<script context="module" lang="ts">
+	import type { Load } from '@sveltejs/kit';
+	import { authLoad } from '$lib/auth';
+
+	export const load: Load = authLoad;
+</script>
+
+<script lang="ts">
+	import type { User } from '$shared/domain';
+	import UserDetails from '$lib/components/UserDetails.svelte';
+
+	export let user: User;
```

Comment:
From whence does the page get the user? I can see that it's a prop but I could not figure it out where the prop is passed to the page.

Reply:
Hah, this is getting harder to see, as I moved `authLoad` function into [a separate file](https://github.com/szattila98/veryrezsi/pull/18/files/ee6d334e1546c1dfc4177d0dfe658331b3ec2bb1#diff-f91399a6444b3e4cd1811221c9e153e17b5ea5f01e1386a517513fd6ccd0c7bb).

Module script (on top of this file) exports  `authLoad` as Load function (SvelteKit docs: https://kit.svelte.dev/docs/loading).
This will run before the page loads (support SSR) and can pass props to a page or layout.
 `authLoad` reads session data and provides _user_  prop.

We might need to add some documentation on this, as it is a maxed SvelteKit magic, what do you think?

Reply:
I think that if we don't add it then I will ask every time this file is changed 😄

Reply:
Added some comments and renemed the function.